posix/fork.c

```diff
@@ -39,98 +52,107 @@ fresetlockfiles (void)
 pid_t
 __libc_fork (void)
 {
+  // printf("calling fork in glibc\n");
+  // int pid = __wasi_fork();
+  struct clone_args args;
+  memset(&args, 0, sizeof(args));
+  args.flags = 0;
+
+  int pid = __clone_internal(&args, NULL, NULL);
+  // int pid = MAKE_SYSCALL(171, "syscall|clone3", NOTUSED, NOTUSED, NOTUSED, NOTUSED, NOTUSED, NOTUSED);
+  return pid;
   /* Determine if we are running multiple threads.  We skip some fork
      handlers in the single-thread case, to make fork safer to use in
      signal handlers.  Although POSIX has dropped async-signal-safe
      requirement for fork (Austin Group tracker issue #62) this is
      best effort to make is async-signal-safe at least for single-thread
      case.  */
-  bool multiple_threads = !SINGLE_THREAD_P;
-  uint64_t lastrun;
+  // bool multiple_threads = !SINGLE_THREAD_P;
+  // uint64_t lastrun;
 
-  lastrun = __run_prefork_handlers (multiple_threads);
+  // lastrun = __run_prefork_handlers (multiple_threads);
 
-  struct nss_database_data nss_database_data;
+  // struct nss_database_data nss_database_data;
 
-  /* If we are not running multiple threads, we do not have to
-     preserve lock state.  If fork runs from a signal handler, only
-     async-signal-safe functions can be used in the child.  These data
-     structures are only used by unsafe functions, so their state does
-     not matter if fork was called from a signal handler.  */
-  if (multiple_threads)
-    {
-      call_function_static_weak (__nss_database_fork_prepare_parent,
-				 &nss_database_data);
+  // /* If we are not running multiple threads, we do not have to
+  //    preserve lock state.  If fork runs from a signal handler, only
+  //    async-signal-safe functions can be used in the child.  These data
+  //    structures are only used by unsafe functions, so their state does
+  //    not matter if fork was called from a signal handler.  */
+  // if (multiple_threads)
+  //   {
```

Comment:
Do we need to remove those lines?

Reply:
I think for the libc code we don' necessarily want to remove the commented original libc code but we want to leave notes for why we are disabling it.